gcpvault.go

```diff
@@ -433,46 +426,71 @@ func newJWTBase(ctx context.Context, cfg Config) (string, error) {
 			Base:   hc.Transport,
 		},
 	}
-	iamClient, err := iam.New(hcIAM)
+
+	claim, err := json.Marshal(map[string]interface{}{
+		"aud": "vault/" + cfg.Role,
+		"sub": serviceAccount,
+		"exp": time.Now().UTC().Add(5 * time.Minute).Unix(),
+	})
+
 	if err != nil {
-		return "", errors.Wrap(err, "unable to init IAM client")
+		return "", errors.Wrap(err, "unable to encode JWT payload")
 	}
 
+	gcpURL := "https://iamcredentials.googleapis.com/v1"
 	if cfg.IAMAddress != "" {
-		iamClient.BasePath = cfg.IAMAddress
+		gcpURL = cfg.IAMAddress
+	}
+	// this additional marshalling is to escape the json
+	payload, err := json.Marshal(string(claim))
+	if err != nil {
+		return "", errors.Wrap(err, "unable to encode JWT payload")
 	}
+	reqBody := []byte("{\"payload\":" + string(payload) + "}")
```

Comment:
A map may be a more readable choice here.

https://play.golang.org/p/IdCg5p9Iwyf
```go
m := map[string]map[string]interface{}{
		"payload": {
			"aud": "vault/" + cfg.Role,
			"sub": serviceAccount,
			"exp": time.Now().UTC().Add(5 * time.Minute).Unix(),
		},
	}
```

Reply:
I am trying to get the payload into this shape. `{"payload":"{\"aud\":\"vault/sadf\",\"exp\":1622572732,\"sub\":\"jp@example.com\"}"}`